from-callback-to-promise/README.md

```diff
@@ -0,0 +1,82 @@
+* **Format:** Mob Programming
+* **Kata:** From Callback to Promise
+* **Where:** [TabCorp](https://www.tabcorp.com.au/)
+* **When:** 19/04/2017 & 26/04/2017
+
+<img src="https://cloud.githubusercontent.com/assets/2061821/25463486/ff38db80-2b39-11e7-986e-61b4f9dc064b.png" width="520px" />
+
+<img src="https://cloud.githubusercontent.com/assets/2061821/25463491/0463ac7a-2b3a-11e7-8f92-a665753fb46a.png" width="520px" />
+
+## Kata: From Callback to Promise
+
+### Description:
+
+Refactoring real code from TabCorp changing Callbacks to Promise using [Mocha](https://mochajs.org/), [Node.js](https://nodejs.org/en/) and [CoffeeScript](http://coffeescript.org/).
+
+### Lessons learned:
+
+1. Testing negative scenarios, you need to reject the promise inside `.then` and writing your expectation inside `.catch`:
+
+```coffeescript
+it 'returns an error', ->
+  myObject.myPromise(parameter)
+    .then -> Promise.reject 'This is not happyily ever after :('
+    .catch (err) ->
+      err.should.be.exactly someError
+```
+
+Before using Callback:
+
+```coffeescript
+it 'returns an error', (done) ->
+  myObject.myFunction parameter, (err) ->
+    err.should.be.exactly someError
+    done()
+```
+
+and we don't need to use `done` since Mocha knows how to handle a Promise.
+
+2. For positive scenarios, you just need to use `.then`:
+
+```coffeescript
+it 'does something', ->
+  myObject.myPromise(parameter).then ->
```

Comment:
Maybe it is worth mentioning that you have to `return` a promise on a mocha test so it waits for the promise's resolution… and because of coffee-script, we don't need an explicit `return` on the last statement…